src/kernl/implementations/linear_layer.py

```diff
@@ -137,52 +137,53 @@ def kernel_fma(
 
     This kernel will consolidate over K
     """
-
-    pid = tl.program_id(axis=0)
+    program_idx = tl.program_id(axis=0)
 
     grid_m = (M + BLOCK_M - 1) // BLOCK_M
     grid_n = (N + BLOCK_N - 1) // BLOCK_N
     # re-order program ID for better L2 performance
     width = GROUP_M * grid_n
-    group_id = pid // width
-    group_size = min(grid_m - group_id * GROUP_M, GROUP_M)
-    pid_m = group_id * GROUP_M + (pid % group_size)
-    pid_n = (pid % width) // (group_size)
+    group_idx = program_idx // width
+    group_size = min(grid_m - group_idx * GROUP_M, GROUP_M)
+    m_block_idx = group_idx * GROUP_M + (program_idx % group_size)
```

Comment:
should it be `block_m_idx` and `block_n_idx` ?

Reply:
done